iceoryx_hoofs/legacy/include/iceoryx_hoofs/design_pattern/builder.hpp

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2023 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_DESIGN_PATTERN_BUILDER_HPP
+#define IOX_HOOFS_DESIGN_PATTERN_BUILDER_HPP
+
+#include "iox/builder.hpp"
+/// @todo iox-#1593 Deprecate include
+/// [[deprecated("Deprecated in 3.0, removed in 4.0, please include 'iox/builder.hpp' instead")]]
```

Comment:
The deprecating mechanism we want to use for the C release `v3.0` relies on deprecating the namespace. I suppose here when using macro, we could go for a simple `#pragma` warning?

Reply:
Good point. I'm not sure what's the best solution here. For now I've replaced that line with a simple Doxygen command for portability. But this doesn't generate a compiler warning... What do you think @mossmaurice and @elfenpiff: `#pragma message("...")` or Doxygen or ???

Reply:
@mossmaurice No hard feelings either way, but `#pragma message` in addition to the comment would not be too bad because it only improves things. Either it works and emits the message (on most compilers) or it does nothing if the compiler does not support it and we are  back to where we are now.

Up to you.

I have no other solution as we have no useful symbols (function names, namespaces etc.) to deprecate here as there are only macros that have no namespace and shall still be usable.

Reply:
Can be made probably even more sophisticated by actually generating some artificial warning by including "bad" code here that does not hurt otherwise, say in an anonymous namespace.

Reply:
> @mossmaurice No hard feelings either way, but `#pragma message` in addition to the comment would not be too bad because it only improves things. Either it works and emits the message (on most compilers) or it does nothing if the compiler does not support it and we are back to where we are now.

@MatthiasKillat I think compilers that don't know the pragma directive even issue a warning which is actually good in this specific case. Therefore I replaced the Doxygen command with `#pragma message` and even found two overlooked old include paths.